packages/dev/core/src/XR/features/WebXRRawCameraAccess.ts

```diff
@@ -0,0 +1,224 @@
+    ) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "camera-access";
+    }
+
+    /**
+     * attach this feature
+     * Will usually be called by the features manager
+     *
+     * @returns true if successful.
+     */
+    public attach(force?: boolean | undefined): boolean {
+        if (!super.attach(force)) {
+            return false;
+        }
+
+        this._glContext = this._xrSessionManager.scene.getEngine()._gl;
+        this._glBinding = new XRWebGLBinding(this._xrSessionManager.session, this._glContext);
+
+        return true;
+    }
+
+    public detach(): boolean {
+        if (!super.detach()) {
+            return false;
+        }
+        this._glBinding = undefined;
+        if (!this.options.doNotDisposeOnDetach) {
+            this._cachedInternalTextures.forEach((t) => t.dispose());
+            this.texturesData.forEach((t) => t.dispose());
+            this._cachedInternalTextures.length = 0;
+            this.texturesData.length = 0;
+            this.cameraIntrinsics.length = 0;
+        }
+        return true;
+    }
+
+    /**
+     * Dispose this feature and all of the resources attached
+     */
+    public dispose(): void {
+        super.dispose();
+        this.onTexturesUpdatedObservable.clear();
+    }
+
+    /**
+     * @see https://github.com/immersive-web/raw-camera-access/blob/main/explainer.md
+     */
+    private _updateCameraIntrinsics(view: XRView, index: number): void {
+        const cameraViewport = {
+            width: view.camera!.width,
+            height: view.camera!.height,
+            x: 0,
+            y: 0,
+        };
+        const p = view.projectionMatrix;
+
+        // Principal point in pixels (typically at or near the center of the viewport)
+        const u0 = ((1 - p[8]) * cameraViewport.width) / 2 + cameraViewport.x;
+        const v0 = ((1 - p[9]) * cameraViewport.height) / 2 + cameraViewport.y;
+
+        // Focal lengths in pixels (these are equal for square pixels)
+        const ax = (cameraViewport.width / 2) * p[0];
+        const ay = (cameraViewport.height / 2) * p[5];
+
+        // Skew factor in pixels (nonzero for rhomboid pixels)
+        const gamma = (cameraViewport.width / 2) * p[4];
+        this.cameraIntrinsics[index] = {
+            u0,
+            v0,
+            ax,
+            ay,
+            gamma,
+            width: cameraViewport.width,
+            height: cameraViewport.height,
+            viewportX: cameraViewport.x,
+            viewportY: cameraViewport.y,
+        };
+    }
+
+    private _updateInternalTextures(view: XRView, index = 0): boolean {
+        if (!view.camera) {
+            return false;
+        }
+        this.viewIndex[index] = view.eye;
+        const lp = this._glBinding?.getCameraImage(view.camera);
+
+        if (!this._cachedInternalTextures[index]) {
+            const internalTexture = new InternalTexture(this._xrSessionManager.scene.getEngine(), InternalTextureSource.Unknown);

```

Comment:
You should pass `true` for the third parameter so that `_hardwareTexture` is not automatically created, else it will produce a leak when you overwrite it.